omod/src/main/java/org/openmrs/module/patientqueue/web/controller/queue/OpdPatientQueueController.java

```diff
@@ -305,11 +305,35 @@ public String selectPatientInSystem(@RequestParam("id") Integer patientId,
 		queue.setVisitStatus("REVISIT");
 		
 		Boolean pd = patient.getDead();
+		
+		//franqq, 24-nov-2015, Issue #8 Patient must pay registration fees before clinician administers treatment
+		Boolean patientPaid = false;
+		Calendar yesterday = Calendar.getInstance();
+		yesterday.add(Calendar.DATE, -1);
+		Calendar today = Calendar.getInstance();
+
+		Concept regFeeConcept = Context.getConceptService().getConcept(3950);
+		ObsService obsService = Context.getObsService();
+		List<Obs> lastObs = obsService.getObservations(Arrays.asList((Person) patient), null,
+				Arrays.asList(regFeeConcept), null, null, null, null, 1, null, yesterday.getTime(), today.getTime(), false);
+		if(lastObs.size() > 0) {
+			patientPaid = true;
+		} else {
+			patientPaid = false;
+		}
+
+		
 		OpdPatientQueue opdPatientQueue = null;
 		if (pd == true) {
 			return "redirect:/module/patientdashboard/main.htm?patientId="
 			+ patientId + "&opdId=" + opdId;
-			} else {
+			} 
+			else if(!patientPaid){
+			String visitstatus = "Not Paid Reg Fee";
```

Comment:
@franqq looks good. I'd recommend one change though, 'magic strings' are usually a bad practice why don't you move the `visitstatus` variable to [OPDPatientQueueConstants.java](https://github.com/bespokeinteractive/patient-queue/blob/version-3.1/api/src/main/java/org/openmrs/module/patientqueue/util/OPDPatientQueueConstants.java) and maybe give it the name `REG_FEE_NOT_PAID_VISIT_STATUS` (feel free to come up with a better name :smile:)